Add attribute `component:navigate`

MasterZydra opened the pull request:
**Usage**
```HTML
<a href="" component:navigate data-component="MyComponent" data-params='{someParam: 42}'>Page 2</a>
```

It replaces the attributes `data-component` and `data-params` in the closest `livewire` tag.

MasterZydra committed:
Add attribute `component:navigate`

MasterZydra commented:
Hi @PhiloNL,
Have you had a chance to take a look at my proposal?

PhiloNL commented:
> Hi @PhiloNL, Have you had a chance to take a look at my proposal?

Not yet, still need to dive into the use case and give this a test.

MasterZydra committed:
Merge branch 'wire-elements:main' into main

LaurinStapf committed:
Add support for Livewire v4 (#46)

## Changes
The `<script>` tag is now rendered via `Blade::render()` and added to
the json response returned by the `/embed` route.
On the client side, `injectLivewire()` consumes this tag and injects the
script into the page.

Since the script is delivered through the embed payload, the helpers
`getLivewireAssetUri()` and `getLivewireUpdateUri()` become obsolete
from my point of view.

LaurinStapf committed:
Migrate Livewire request hook and fix component class resolution (#47)

## Changes
- Replace deprecated `Livewire.hook('request')` with
`Livewire.interceptRequest`
- Send `X-Wire-Extender` header with a non-empty value to avoid
potential
  stripping by proxies/load balancers that ignore empty custom headers.
- Fix component class resolution by avoiding `$this` usage in static
context

LaurinStapf committed:
Fix typo (#48)

MasterZydra commented:
Hi, do you still have this PR in mind 😉?

MasterZydra committed:
Merge branch 'wire-elements:2.x' into main
